web/frontend/src/api.ts

```diff
@@ -31,40 +56,58 @@ import genericApi from './gen/proto/api/component/generic/v1/generic_pb.esm';
 import gripperApi from './gen/proto/api/component/gripper/v1/gripper_pb.esm';
 import inputControllerApi from './gen/proto/api/component/inputcontroller/v1/input_controller_pb.esm';
 import motorApi from './gen/proto/api/component/motor/v1/motor_pb.esm';
+import motionApi from './gen/proto/api/service/motion/v1/motion_pb.esm';
 import movementSensorApi from './gen/proto/api/component/movementsensor/v1/movementsensor_pb.esm';
+import navigationApi from './gen/proto/api/service/navigation/v1/navigation_pb.esm';
 import robotApi from './gen/proto/api/robot/v1/robot_pb.esm';
 import sensorsApi from './gen/proto/api/service/sensors/v1/sensors_pb.esm';
 import servoApi from './gen/proto/api/component/servo/v1/servo_pb.esm';
+import slamApi from './gen/proto/api/service/slam/v1/slam_pb.esm';
 import streamApi from './gen/proto/stream/v1/stream_pb.esm';
 import visionApi from './gen/proto/api/service/vision/v1/vision_pb.esm';
 
-/**
- * Every window variable on this page is being currently used by the blockly page in App.
- * Once we switch blockly to using import / export we should remove / clean up these window variables.
- */
-
-window.commonApi = commonApi;
-window.armApi = armApi;
-window.baseApi = baseApi;
-window.boardApi = boardApi;
-window.cameraApi = cameraApi;
-window.gantryApi = gantryApi;
-window.genericApi = genericApi;
-window.gripperApi = gripperApi;
-window.movementSensorApi = movementSensorApi;
-window.inputControllerApi = inputControllerApi;
-window.motorApi = motorApi;
-window.sensorsApi = sensorsApi;
-window.servoApi = servoApi;
-window.streamApi = streamApi;
-window.visionApi = visionApi;
-
-/**
- * This window variable is used by the config page to access the discovery service.
- * As with variables above, once we switch to using import / export we should
- * remove / clean up these window variables.
- */
-window.robotApi = robotApi;
+export {
+  commonApi,
+  armApi,
+  baseApi,
+  boardApi,
+  cameraApi,
+  gantryApi,
+  genericApi,
+  gripperApi,
+  inputControllerApi,
+  motorApi,
+  motionApi,
+  movementSensorApi,
+  navigationApi,
+  robotApi,
+  sensorsApi,
+  servoApi,
+  slamApi,
+  streamApi,
+  visionApi
+};
+
+let transport: grpc.TransportFactory;
+
+export const createStreamService = () => new StreamServiceClient(window.webrtcHost, { transport });
```

Comment:
doing it this way will break reconnect unfortunately. my work on a session manager has me creating a pre-SDK concept of a light connection/client manager that will avoid this issue but also allow for us to reintroduce this effort of on demand creation

Reply:
Sounds good 👍 I'll close this one then in light of the work you're doing